athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -296,6 +291,25 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
         return new GetSplitsResponse(request.getCatalogName(), splits);
     }
 
+    /**
+     * Get get all data streams from ES, 1 data stream can contains multiple indices which start with ".ds-xxxxxxxxxx"
+     * return empty if not supported.
+     * Notes: AWS's version of ES doesn't support data stream, it is a feature only available in the Elasticsearch distributed by Elastic itself, licensed under the Elastic License.
+     */
+    private Stream<String> getDataStreamNames(AwsRestHighLevelClient client)
+    {
+        try {
+            return client.indices().getDataStream(new GetDataStreamRequest("*"), RequestOptions.DEFAULT).getDataStreams()
+                    .stream()
+                    .map(dataStream -> dataStream.getName());
+        }
+        // gracefully exit for exception or non-support data stream.
+        catch (Exception ex) {
+            logger.warn("getDataStreamNamesFromClient: Unable to retrieve datastream or cluster version not support data stream, ignore datastream, message: {}.", ex.getMessage(), ex);
```

Comment:
If you want to pass in the exception object itself, then it's not worthwhile to pass in the message separately. when slf4j logs the warning, it will log the exception's message and stack trace. So you should just write this:

Notice that I also updated the method name you write (which. you updated in the method header but not here)
```
logger.warn("getDataStreamNames: Unable to receive data stream, or the cluster version does not support data stream.", ex);
```

Also, what is the exact exception that's thrown? Can we make this catch block more specific?